### PR TITLE
chore: docs navigation

### PR DESCRIPTION
**Pull request by p7nov**
Set some order on the docs navigation pane and index page.

### CODE REVIEW

File: data/_nav.yml [hunk 1]
```diff
@@ -179,15 +184,16 @@ reference:
...
+      description: Kotlin Coroutines Library
+      external:
+        path: kotlinx.coroutines/docs
+        nav: _nav.yml
+        base: /docs/reference/coroutines
+        repo: https://github.com/kotlin/kotlinx.coroutines
+        branch: master
 
-    - title: Other
+    - title: Various Language Features
```
Comment (abreslav) [inline data/_nav.yml]: Looks like we have a problem here: this group of items has no meaningful name and the user will never guess what went under "Various language features". Please think about possible solutions to this, I'm happy to discuss options if needed

File: data/_nav.yml [hunk 2]
```diff
@@ -215,23 +221,30 @@ reference:
...
-    - title: Core Libraries
+    - title: Core Libraries Reference
       content:
         - url: /api/latest/jvm/stdlib/index.html
           title: Standard Library
         - url: /api/latest/kotlin.test/index.html
           title: kotlin.test
 
-    - title: Reference
+    - title: Language Reference
```
Comment (abreslav) [inline data/_nav.yml]: The contents of this group are not what's conventionally called a language reference. This is another problem of discoverability. Please propose solutions to it

File: data/_nav.yml [hunk 3]
```diff
@@ -215,23 +221,30 @@ reference:
...
+    - title: Multiplatform Programming
+      description: Reusing code across JVM, JS, and Native
+      content:
+        - url: /docs/reference/platform-specific-declarations.html
+          title: Platform-Specific Declarations
+        - url: /docs/reference/building-mpp-with-gradle.html
+          title: Building with Gradle
 
-    - title: Java Interop
+    - title: Java
```
Comment (abreslav) [inline data/_nav.yml]: I'm afraid that a user can't guess what this title means. Java is a programming language (so is JavaScript), and this is a Kotlin website...

File: pages/docs/reference/index.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ title: "Reference"
 <div style="display: inline-flex; border: 1px solid rgb(229, 229, 229); border-radius: 40px;" >
  <div id="material" onmouseover="this.style.backgroundColor='#F0F0F0'" onmouseout="this.style.backgroundColor=''" style="cursor: pointer; padding: 1rem 2rem;border-radius: 40px; border: 1px solid transparent;" >All Materials</div>
  <div id="start" onmouseover="this.style.backgroundColor='#F0F0F0'" onmouseout="this.style.backgroundColor=''" style="cursor: pointer; padding: 1rem 2rem;border-radius: 40px;">Getting Started</div>
- <div id="migrate" onmouseover="this.style.backgroundColor='#F0F0F0'" onmouseout="this.style.backgroundColor=''" style="cursor: pointer; padding: 1rem 2rem;border-radius: 40px;">Migrating from Java</div>
+ <div id="migrate" onmouseover="this.style.backgroundColor='#F0F0F0'" onmouseout="this.style.backgroundColor=''" style="cursor: pointer; padding: 1rem 2rem;border-radius: 40px;">Migration</div>
```
Comment (abreslav) [inline pages/docs/reference/index.md]: I'm afraid that "Migration" can be ambiguous in this context. How about "Migrating to Kotlin"?

**Comment by p7nov** [comment]
Closing after discussing the scope of desired changes.

**Pull request closed without merging**